eng/tools/generate-dev-builds/index.js

```diff
@@ -0,0 +1,218 @@
+const readFile = util.promisify(fs.readFile);
+const parse = require(`${__dirname}/../../../common/lib/jju/parse`).parse;
+const semver = require('semver');
+
+const appendPackageData = (data, pkgSrc, pkgJson) => {
+
+  var dep = new Map();
+  for(var key in pkgJson.dependencies){
+    dep.set(key, pkgJson.dependencies[key]);
+  }
+
+  var devDep = new Map();
+  for(var key in pkgJson.devDependencies){
+    devDep.set(key, pkgJson.devDependencies[key]);
+  }
+
+  data[pkgJson.name] = {
+    src: pkgSrc,
+    ver: pkgJson.version,
+    dep: dep,
+    devDep: devDep,
+    visited: false,
+    visitedDep : false
+  };
+};
+
+const getRushPackages = async (rushPath) => {
+  const baseDir = path.dirname(rushPath);
+  const rushJson = parse(await readFile(rushPath, "utf8"));
+  const packageData = {};
+
+  for (const proj of rushJson.projects) {
+    const projDir = path.join(baseDir, proj.projectFolder);
+    const packageJson = parse(await readFile(path.join(projDir, "package.json"), "utf8"));
+    appendPackageData(packageData, projDir, packageJson);
+  }
+
+  return packageData;
+};
+
+async function writeToPackageJson(pckJsonPath,contents){
+  console.log("Inside writeTo packageJson function for - " + pckJsonPath );
+  var data = JSON.stringify(contents,null,2);
+  try{
+      fs.access(pckJsonPath, fs.F_OK, (err) => {
+      if (err) {
+        console.error(err);
+        return;
+      }
+    });
+
+    await fs.writeFile(pckJsonPath, data, (err) => {
+      if(err){
+        console.error(err);
+        return;
+      }
+    });
+    console.info("File " + pckJsonPath + " created successfully with Node.js v10 fs/promises!");
+  }
+  catch(e){
+    console.error(`\n\n${e.toString()}\n\n`);
+    process.exit(1);
+  }
+  
+}
+
+function editPackageJson(rushPackages){
+    for(var package in rushPackages){
+     var pckJsonPath = rushPackages[package].src+"/"+"package.json";
+     console.log("attempting to read..");
+     console.log(pckJsonPath);
+     var contents = JSON.parse(fs.readFileSync(pckJsonPath).toString());
+     contents.version = rushPackages[package].ver;
+
+    //Checking dependencies..
+    for(var [dep,version] of rushPackages[package].dep){
+      console.log("checking..."+ dep + ":" + version);
+      if(rushPackages[dep]){
+        contents.dependencies[dep] = version;
+      }
+    }
+
+    //Checking devDependencies..
+    for(var [devDep,version] of rushPackages[package].devDep){
+      console.log("checking..."+ devDep + ":" + version);
+      if(rushPackages[devDep]){
+        contents.devDependencies[devDep] = version;
+      }
+    }
+    writeToPackageJson(pckJsonPath,contents);
+  }
+}
+
+function editVersion(rushPackages, buildNo,package){
+  console.log("Inside editVersion function - "+ package);
+  rushPackages[package].ver = rushPackages[package].ver+ "-dev-"+buildNo;
+  return rushPackages;
+}
+
+function editDependencyVersion(rushPackages,package,dep,version,depType){
```

Comment:
Did we get a answer to whether or not we want to update the dependencies as part of this work or not? I'm still thinking we don't want to do that as part of our daily dev builds.

Reply:
removed the dependency version updates from the code